azure/services/availabilitysets/availabilitysets_test.go

```diff
@@ -214,13 +224,13 @@ func TestDeleteAvailabilitySets(t *testing.T) {
 		},
 		{
 			name:          "error in getting availability set",
-			expectedError: "failed to get availability set test-as in resource group test-rg: #: Internal Server Error: StatusCode=500",
+			expectedError: "failed to get availability set test-as in resource group test-rg:.*#: Internal Server Error: StatusCode=500",
 			expect: func(s *mock_availabilitysets.MockAvailabilitySetScopeMockRecorder, m *mock_async.MockGetterMockRecorder, r *mock_async.MockReconcilerMockRecorder) {
 				s.AvailabilitySetSpec().Return(&fakeSetSpec)
 				gomock.InOrder(
 					s.DefaultedAzureServiceReconcileTimeout().Return(reconciler.DefaultAzureServiceReconcileTimeout),
-					m.Get(gomockinternal.AContext(), &fakeSetSpec).Return(nil, internalError),
-					s.UpdateDeleteStatus(infrav1.AvailabilitySetReadyCondition, serviceName, gomockinternal.ErrStrEq("failed to get availability set test-as in resource group test-rg: #: Internal Server Error: StatusCode=500")),
+					m.Get(gomockinternal.AContext(), &fakeSetSpec).Return(nil, internalError()),
```

Comment:
Maybe `gomockinternal.ErrStrEq(internalError().Error())` would work here? I could try that if we want to preserve the specificity of this mock call.

Reply:
If that works, I'm +1. If it doesn't I wouldn't worry about digging much deeper for now.

Reply:
That did work; I updated the three uses of `gomock.Any()`.